### ISSUE TITLE
iOS: "java.io.IOException: Couldn't get lock for" while create instance of FileHandler

### ISSUE THREAD

**Issue by StasKalishenko** [google/j2objc#1226]
Hi! I'm trying to add logging to file in iOS (in Android it works fine), but when I create an instance of FileHandler, get IOException telling that "Couldn't get lock for [my absolute path to log file]". What I'm doing wrong?
Problematic piece of code:
FileHandler handler = new FileHandler(getFilePath(name), MAX_LOG_FILE_SIZE,
				MAX_LOG_FILE_COUNT, true);
P.S.: I checked the existence of all folders in the path - everything is fine.

**Comment by tomball** [comment]
[Here's where the exception is thrown](https://github.com/google/j2objc/blob/master/jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/logging/FileHandler.java#L390), which suggests that your app is opening more than 100 (MAX_LOCKS) log files. That sounds unlikely, but if that's true then remember your iOS device isn't a server. :-)

More likely is that this is [JDK-8153955](https://bugs.openjdk.java.net/browse/JDK-8153955), which is fixed in Java 8/Android API 26 (Oreo). However, this fix depends upon issue #1064 (add java.nio.file package, also introduced in Oreo).

**Comment by StasKalishenko** [comment]
Am I right, this fix [https://github.com/google/j2objc/issues/1064] is in progress?

**Comment by tomball** [comment]
Yes, but implementing a new file system API correctly is a big task, especially when it requires new native code. So we're looking at weeks, not hours.

In the meantime, consider setting MAX_LOG_FILE_COUNT to be less than [FileHandler's MAX_LOCKS](https://github.com/google/j2objc/blob/master/jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/logging/FileHandler.java#L130) count of 100. [Needing more than 100 locks is what caused the IOException to be thrown.](https://github.com/google/j2objc/blob/master/jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/logging/FileHandler.java#L390)
